core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1556,6 +1681,133 @@ public void warmUp() {
+                }
+            }
+        } finally {
+            Misc.free(detachedMetadata);
+            // remove _meta
+            if (!ff.remove(other2)) {
+                LOG.error().$("cannot remove [errno=").$(ff.errno())
+                        .$(", path=").$(other2)
+                        .I$();
+            }
+        }
+
+        // load/check _cv, updating local column tops
+        // set current _cv to where the partition was
+        other2.parent().concat(COLUMN_VERSION_FILE_NAME).$();
+        if (!ff.exists(other2)) {
+            // Backups and older versions of detached partitions will not have _cv
+            LOG.info().$("detached _cv not found, skipping check [path=").$(other2).I$();
+            return;
+        }
+        try {
+            if (detachedColumnVersionReader == null) {
+                detachedColumnVersionReader = new ColumnVersionReader();
+            }
+            detachedColumnVersionReader.ofRO(ff, other2);
+            detachedColumnVersionReader.readSafe(MillisecondClockImpl.INSTANCE, Long.MAX_VALUE);
+            LongList entries = columnVersionWriter.getCachedList();
+            LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+            for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
+                // detachedEntries: [partitionTimestamp, columnIndex, columnNameTxn, columnTop]*
+                long columnTop = detachedEntries.getQuick(i + 3);
+                if (timestamp == detachedEntries.getQuick(i) && columnTop != entries.getQuick(i + 3)) {
+                    columnVersionWriter.upsert(
+                            timestamp,
+                            (int) detachedEntries.getQuick(i + 1),
+                            detachedEntries.getQuick(i + 2),
+                            columnTop
+                    );
+                }
+            }
+        } finally {
+            Misc.free(detachedColumnVersionReader);
+            // remove _cv
+            if (!ff.remove(other2)) {
+                LOG.error().$("cannot remove [errno=").$(ff.errno())
+                        .$(", path=").$(other2)
+                        .I$();
+            }
+        }
+    }
+
+    private void commitDetachPartition(long timestamp, long minTimestamp) {
+        // when we want to delete first partition we must find out
+        // minTimestamp from next partition if it exists or next partition and so on
+        //
+        // when somebody removed data directories manually and then
+        // attempts to tidy up metadata with logical partition delete
+        // we have to uphold the effort and re-compute table size and its minTimestamp from
+        // what remains on disk
+
+        // find out if we are removing min partition
+        long nextMinTimestamp = minTimestamp;
+        if (timestamp == txWriter.getPartitionTimestamp(0)) {
+            nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
```

Comment:
This sets that column is not present in the partition.

Instead we want to erase any information about this partition in Column Version file. Please add method to remove all records about the partition in the Column Version file and use it here.

Reply:
I have added a new method to CVW:
```
public void remove(long partitionTimestamp, int columnIndex) {
        int recordIndex = getRecordIndex(partitionTimestamp, columnIndex);
        if (recordIndex >= 0) {
            getCachedList().removeIndexBlock(recordIndex, BLOCK_SIZE);
            hasChanges = true;
        }
    }
```
which deletes the information about the detached partition.